qiskit_ibm_runtime/qiskit_runtime_service.py

```diff
@@ -927,6 +930,36 @@ def _run(self, *args: Any, **kwargs: Any) -> Union[RuntimeJob, RuntimeJobV2]:
         """Private run method"""
         return self.run(*args, **kwargs)
 
+    def check_pending_jobs(self) -> None:
+        """Check the number of pending jobs and wait for the oldest pending job if
+        the maximum number of pending jobs has been reached.
+        """
+        try:
+            usage = self.usage().get("byInstance")[0]
+            pending_jobs = usage.get("pendingJobs")
+            max_pending_jobs = usage.get("maxPendingJobs")
+            if pending_jobs >= max_pending_jobs:
+                logger.warning(
+                    "The maximum number of pending jobs on the open plan is %s. "
+                    "You currently have %s pending jobs. Waiting for a pending job to complete.",
+                    max_pending_jobs,
+                    pending_jobs,
+                )
+                oldest_running = self.jobs(limit=1, descending=False, pending=True)
+                if oldest_running:
+                    try:
+                        oldest_running[0].wait_for_final_state(timeout=300)
```

Comment:
Waiting for 5 minutes may be too long for users. It would be helpful to include the time we will wait (worst case) in the warning message for pending jobs.